src/cmd/troubleshoot/check_dynakube.go

```diff
@@ -0,0 +1,268 @@
+	return nil
+}
+
+func checkApiUrl(apiReader client.Reader, troubleshootContext *TestData) error {
+	tslog.Infof("checking if api url is valid...")
+
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+
+	apiUrl := dynakube.Spec.APIURL
+
+	if len(apiUrl) < 4 || apiUrl[len(apiUrl)-4:] != "/api" {
+		tslog.Errorf("api url has to end on '/api' (%s)", apiUrl)
+		return fmt.Errorf("api url has to end on '/api' (%s)", apiUrl)
+	}
+
+	tslog.Infof("api url correctly ends on '/api'")
+
+	tslog.Infof("api url is valid")
+	return nil
+}
+
+func checkDynatraceApiSecret(apiReader client.Reader, troubleshootContext *TestData) error {
+	tslog.Infof("checking if secret is valid ...")
+
+	// use dynakube name or tokens value if set
+	troubleshootContext.dynatraceApiSecretName = troubleshootContext.dynakubeName
+
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+
+	if dynakube.Spec.Tokens != "" {
+		troubleshootContext.dynatraceApiSecretName = dynakube.Spec.Tokens
+	}
+	return nil
+}
+
+func checkIfDynatraceApiSecretWithTheGivenNameExists(apiReader client.Reader, troubleshootContext *TestData) error {
+	secret := corev1.Secret{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynatraceApiSecretName, Namespace: troubleshootContext.namespaceName}, &secret); err != nil {
+		tslog.WithErrorf(err, "'%s:%s' secret is missing", troubleshootContext.namespaceName, troubleshootContext.dynatraceApiSecretName)
+		return err
+	}
+	tslog.Infof("'%s:%s' secret exists", troubleshootContext.namespaceName, troubleshootContext.dynatraceApiSecretName)
+	return nil
+}
+
+func checkIfDynatraceApiSecretHasApiToken(apiReader client.Reader, troubleshootContext *TestData) error {
+	secret := corev1.Secret{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynatraceApiSecretName, Namespace: troubleshootContext.namespaceName}, &secret); err != nil {
+		tslog.WithErrorf(err, "'%s:%s' secret is missing", troubleshootContext.namespaceName, troubleshootContext.dynatraceApiSecretName)
+		return err
+	}
+
+	apiTokenByte, ok := secret.Data["apiToken"]
+	if !ok {
+		tslog.Errorf("token apiToken does not exist in '%s:%s' secret", troubleshootContext.namespaceName, troubleshootContext.dynatraceApiSecretName)
+		return fmt.Errorf("token apiToken does not exist in secret")
```

Comment:
new errors need to be instantiated with `errors.New`, see the contribution guidelines for reference
```suggestion
		return errors.New("token apiToken does not exist in secret")
```